plugins/postcss-lab-function/src/on-css-function.ts

```diff
@@ -68,10 +70,80 @@ function onCSSFunction(node: FunctionNode) {
 		...channelNode3,
 		value: String(rgbValues[2]),
 	});
-
 }
 
-export default onCSSFunction;
+export function onCSSFunctionDisplayP3(node: FunctionNode, decl: Declaration, result: Result, preserve: boolean) {
+	const originalForWarnings = valueParser.stringify(node);
+
+	const value = node.value;
+	const rawNodes = node.nodes;
+	const relevantNodes = rawNodes.slice().filter((x) => {
+		return x.type !== 'comment' && x.type !== 'space';
+	});
+
+	let nodes: Lch | Lab | null = null;
+	if (value === 'lab') {
+		nodes = labFunctionContents(relevantNodes);
+	} else if (value === 'lch') {
+		nodes = lchFunctionContents(relevantNodes);
+	}
+
+	if (!nodes) {
+		return;
+	}
+
+	if (relevantNodes.length > 3 && (!nodes.slash || !nodes.alpha)) {
+		return;
+	}
+
+	// rename the Color function to `color`
+	node.value = 'color';
+
+	/** Extracted Color channels. */
+	const [channelNode1, channelNode2, channelNode3] = channelNodes(nodes);
+	const [channelDimension1, channelDimension2, channelDimension3] = channelDimensions(nodes);
+
+	/** Corresponding Color transformer. */
+	const toDisplayP3 = value === 'lab' ? labToDisplayP3 : lchToDisplayP3;
+
+	/** RGB channels from the source color. */
+	const channelNumbers: [number, number, number] = [
+		channelDimension1.number,
+		channelDimension2.number,
+		channelDimension3.number,
+	].map(
+		channelNumber => parseFloat(channelNumber),
+	) as [number, number, number];
+
+	const [rgbValues, inGamut] = toDisplayP3(
+		channelNumbers,
+	);
+
+	if (!inGamut && preserve) {
+		decl.warn(
+			result,
+			`"${originalForWarnings}" is out of gamut for "display-p3". When "preserve: true" is set this will lead to unexpected results in some browsers.`,
```

Comment:
```suggestion
			`"${originalForWarnings}" is out of gamut for "display-p3". Given "preserve: true" is set this will lead to unexpected results in some browsers.`,
```